docs/proposals/formatted-output.md

```diff
@@ -0,0 +1,226 @@
+
+The formatted output is not intended to supersede the prettified human-readable and friendly output text of ORAS CLI. It aims to provide programming-friendly experience for developers who want to automate their workflows on machines especially on Unix and most Unix-like operating systems, without parsing unstructured text. It will increase the developer experience for ORAS in automation and scripting scenarios. 
+
+## Scenarios
+
+### Scripting
+
+Alice is a developer who wants to batch operations with ORAS in her Shell script. In order to automate some routine workflow in containers secure supply chain scenario, she wants the machine to get the image digest from the JSON output objects that are emitted by `oras push`, then use utility like [xargs](https://en.wikipedia.org/wiki/Xargs) to enable an ORAS command to act on the output of another command and perform further steps. In this way, she can chain commands together. For example, she can use `oras attach` to attach an SBOM to the image using its image digest as a argument outputted from `oras push`.
+
+For example, push an artifact to a registry and generate the artifact reference in the standard output, then attach an SBOM to the artifact using the artifact reference (`$REG/$REPO:$DIGEST`) outputted from the first command, finally sign the attached SBOM with another tool against the SBOM file's reference (`$REG/$REPO:$DIGEST`) outputted from the last step.
+
+```shell
+oras push $REG/$REPO:$TAG hello.txt --format '{{.Reference}}' |\
+xargs -I _ oras attach --artifact-type sbom/example _ sbom.spdx --format '{{.Reference}}' |\
+xargs -I _ notation sign _ 
+```
+
+### CI/CD
+
+Bob is a DevOps engineer. He uses the ORAS GitHub Actions [Setup action](https://github.com/oras-project/setup-oras) to install ORAS in his CI/CD workflow. He wants to chain multiple ORAS commands in a Shell script to perform multiple operations.
+
+```yaml
+jobs:
+  example-job:
+    steps:
+      - uses: oras-project/setup-oras@v1
+      - run: |
+          oras pull $REG/$REPO:$TAG --format '{{.Files[0].Path}}' |\
+          xargs -I _ docker import _
+```
+
+## Proposal: support formatting output as JSON
+
+- Use the `--format json` flag to change the default human-readable prettified output to machine-readable raw JSON. 
+- Use the `--format` with Go template to custom the output fields.
+
+## Use cases
+
+### oras pull 
+
+Pull an artifact and display its metadata as formatted JSON in standard output. The following fields should be formatted in a JSON output:
+
+- `reference`: reference of the image manifest uses in the `oras pull` command
```

Comment:
I think we should use `artifact` instead of `image` and user won't need to know if it's pointing to an artifact or artifact manifest.
```suggestion
- `reference`: full artifact reference by digest
```